lib/taste_tester/commands.rb

```diff
@@ -210,5 +210,88 @@ def self.upload
       logger.error(exception.backtrace.join("\n"))
       exit 1
     end
+
+    def self.impact
+      if TasteTester::Config.json
+        logger.error('JSON output format is not yet implemented')
+        exit(1)
+      end
+
+      # Use the repository specified in config.rb to calculate the changes
+      # that may affect Chef. These changes will be further analyzed to
+      # determine specific roles which may change due to modifed dependencies.
+      repo = BetweenMeals::Repo.get(
+        TasteTester::Config.repo_type,
+        TasteTester::Config.repo,
+        logger,
+      )
+      if repo && !repo.exists?
+        fail "Could not open repo from #{TasteTester::Config.repo}"
+      end
+
+      changes = _find_changeset(repo)
+
+      # Use Knife (or custom logic) to check the dependencies of each role
+      # against the list of changes. `impacted_roles` will contian the set
+      # of roles with direct or indirect (dependency) modifications.
+      impacted_roles = TasteTester::Hooks.impact_find_roles(changes)
+
+      # Do any post processing required on the list of impacted roles, such
+      # as looking up hostnames associated with each role.
+      final_impact = TasteTester::Hooks.post_impact(impacted_roles)
+
+      # Print the calculated impact. If a print hook is defined that
+      # returns true, then the default print function is skipped.
+      unless TasteTester::Hooks.print_impact(final_impact)
```

Comment:
we should be consistent here. Either we do this one like you did `impact_find_roles` above, where the default code is in the base plugin **or** we do `find_impact_roles` the way we did this one, where if the plugin is a no-op we call an internal one...  I *think* I prefer this one to the other one, but I don't feel strongly.

Reply:
When you say "this one," do you mean having a no-op that triggers a private, default function?  I think it probably makes sense to have the actual functionality stay in the commands.rb file and have hooks.rb be stubs.

Reply:
Yes, exactly :)